packages/react/src/generators/library/lib/create-files.ts

```diff
@@ -68,10 +68,24 @@ export function createFiles(host: Tree, options: NormalizedSchema) {
     });
   }
 
-  if (!options.publishable && !options.buildable) {
+  if (
+    !options.publishable &&
+    !options.buildable &&
+    !options.isUsingTsSolutionConfig
+  ) {
     host.delete(`${options.projectRoot}/package.json`);
   }
 
+  if (options.isUsingTsSolutionConfig) {
```

Comment:
It would be clearer to do this where we add the project configuration to `project.json`, so the two ways to add the Nx project configuration are side by side.